src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -88,6 +88,23 @@ public override string ToString()
     }
 }";
 
+        [Fact, WorkItem(26810, "https://github.com/dotnet/roslyn/issues/26810")]
+        public void LockStatement()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    void F(object? maybeNull)
+    {
+        lock (maybeNull) { }
```

Comment:
Please add some more tests:
* Non-null locked object
* `Unknown?` locked object
* Locked object defined in an oblivious context